xbmc/cores/DllLoader/exports/emu_msvcrt.cpp

```diff
@@ -1633,7 +1631,7 @@ extern "C"
     int ret;
 
     ret = dll_fgetpos64(stream, &tmpPos);
-#if !defined(TARGET_POSIX) || defined(TARGET_DARWIN) || defined(TARGET_FREEBSD) || defined(TARGET_ANDROID)
+#if !defined(__GLIBC__) || defined(TARGET_DARWIN) || defined(TARGET_FREEBSD) || defined(TARGET_ANDROID)
```

Comment:
I hope my following assumptions are correct, my understand of the code is limited.

Musl follows the POSIX standard, the patch I posted tinkers around with the internal representation of different archs. TARGET_POSIX is defined by Kodi configure in m4/xbmc_arch.m4, based on this patchset this will also the case for musl toolchains but the patch author seems to want to use the non-POSIX code path in emu_msvcrt.cpp for whatever reason.

The problem is that Musl does not provide a macro like \__MUSL\__ to identify it:
http://wiki.musl-libc.org/wiki/FAQ#Q:_why_is_there_no_MUSL_macro_.3F
so other ways need to be found to treat Musl differently from glibc/uClibc. What about adding an internal define to m4/xbmc_arch.m4 like we do for FreeBSD, something like -D_MUSL
```
  i*86*-linux-gnu*|i*86*-*-linux-uclibc*|i*86*-*-linux-musl*)
     AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_MUSL")
```
and patch emu_msvcrt.cpp like this?
```
-#if !defined(TARGET_POSIX) || defined(TARGET_DARWIN) || defined(TARGET_FREEBSD) || defined(TARGET_ANDROID)
+#if !defined(TARGET_POSIX) || defined(TARGET_DARWIN) || defined(TARGET_FREEBSD) || defined(TARGET_ANDROID) || defined(_MUSL)
```